src/main/scala/com/bwsw/kafka/reader/MessageQueue.scala

```diff
@@ -42,17 +44,21 @@ class MessageQueue[K,V](consumer: Consumer[K,V]) {
     * @param n count of InputEnvelope entities to extract
     */
   def take(n: Int): List[InputEnvelope[V]] = {
+    logger.trace(s"take(n: $n)")
     if (buffer.size < n) {
+      logger.debug("Count of entities in buffer is not enough, new entities will be received from Kafka")
       fill()
     }
 
     val sizeAfterFill = buffer.size
+    logger.debug(s"Count of entities in buffer after receiving from Kafka is: $sizeAfterFill")
 
     val envelopes = if (sizeAfterFill < n) {
       buffer.take(sizeAfterFill)
     } else {
       buffer.take(n)
     }
+    logger.debug(s"Entities: $envelopes received")
```

Comment:
The following entities: $envelopes will be returned

Reply:
ok